### ISSUE TITLE
`store_container_labels` option conflicts with `docker_env_metadata_whitelist`

### ISSUE THREAD

**Issue by Colstuwjx** [google/cadvisor#2327]
Hi,

I'm using cadvisor, in order to ignore some labels which we really don't care, I have taken `store_container_labels` and `whitelisted_container_labels` to make cadvisor ONLY exposes whitelisted container labels, it works fine.

But another requirement comes in, and things broken. we want to expose some docker environment as cadvisor prom labels, and we found `docker_env_metadata_whitelist` option, but we strangely find that it didn't work as expect. 

After deep into the code, I found that the [DefaultContainerLabels](https://github.com/google/cadvisor/blob/master/metrics/prometheus.go#L1144) function implements `docker_env_metadata_whitelist` feature, and [BaseContainerLabels](https://github.com/google/cadvisor/blob/master/metrics/prometheus.go#L1167)  loss it and implements another `store_container_labels` feature.

Maybe we need to also implements `docker_env_metadata_whitelist` feature inside `BaseContainerLabels` function?

Thanks.

**Comment by dashpole** [comment]
Yeah, it looks like we just skip env vars all-together.  We should probably add those.  Feel free to open a PR for that.

**Comment by alyssahardy** [comment]
I'm interested in this fix as well.

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
